enhancements/single-node-production-edge-cluster-profile.md

```diff
@@ -0,0 +1,380 @@
+
+#### Examples
+
+These are generalized examples to consider, in addition to the aforementioned [maturity levels][maturity-levels].
+
+##### Dev Preview -> Tech Preview
+
+- Ability to utilize the enhancement end to end
+- End user documentation, relative API stability
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+##### Tech Preview -> GA
+
+- More testing (upgrade, downgrade, scale)
+- Sufficient time for feedback
+- Available by default
+
+**For non-optional features moving to GA, the graduation criteria must include
+end to end tests.**
+
+##### Removing a deprecated feature
+
+- Announce deprecation and support policy of the existing feature
+- Deprecate the feature
+
+### Upgrade / Downgrade Strategy
+
+In-place upgrades and downgrades will not be supported for this first 
+iteration. Upgrades will initially only be achieved by redeploying
+the machine and its workload.
+
+### Version Skew Strategy
+
+With only one node and no in-place upgrade, there will be no
+version skew.
+
+## Implementation History
+
+Major milestones in the life cycle of a proposal should be tracked in `Implementation
+History`.
+
+## Drawbacks
+
+1. Clusters built using this profile will not have many of the high-availability
+   features that OpenShift users have come to rely on. We will need to communicate
+   the relevant limitations and the approaches to deal with them clearly.
+
+## Alternatives
+
+### Single-node deployments based on static pods
+
+[Enhancement proposal
+302](https://github.com/openshift/enhancements/pull/302) describes an
+approach for creating the manifests to run a set of static pods to run
+the cluster control plane, instead of using operators.
+
+[Enhancement proposal
+440](https://github.com/openshift/enhancements/pull/440) builds on 302
+and describes another approach for creating a single-node deployment
+by having the installer create an Ignition configuration file to
+define static pods for the control plane services.
+
+Either approach may be useful for more constrained environments, but
+the effort involved - including the effort required to make any
+relevant optional software available for this deployment type - 
+is not obviously worth the resource savings in the less
+resource-contrained environments addressed this proposal.
+
+### "Remote workers" for widely-dispersed clusters
+
+A "remote worker" approach - where the worker nodes are separated
+from the control plane by significant (physical or network
+topological) distance - is appealing because it has the benefit
+of reducing the per-site control plane overhead demanded by
+autonomous edge clusters.
+
+However, there are drawbacks related to what happens
+when those worker nodes lose communication with the cluster control
+plane. The most significant problem is that if a node reboots while
+it has lost communication with the control plane, it does not
+restart any pods it was previously running until communication
```

Comment:
Hmm...this seems like it shouldn't be too hard to fix.  Just start daemonsets by default for example, and add an annotation to pods to do so?

Reply:
(Although one problem right now is that crio currently wipes all containers across minor version upgrades...)

Reply:
The bigger problem is with user workloads, especially any that also need to talk to the API to learn what to do (fetch a ConfigMap, etc.). It's likely possible to make all of that work, but at some point we would be working around kubernetes instead of taking advantage of it, and the application deployment for the remote workers would be different from centralized sites which means we don't meet the goal of having application management be as uniform as possible.

There's more background about why this approach was rejected in some internal documents, which we're working on publishing more widely as more people get involved in examining these requirements.